### PR TITLE
Improv/extended autoplay types etc

### PR DESCRIPTION
**Pull request by cjpillsbury**

### CODE REVIEW

File: packages/mux-player-react/src/index.tsx [hunk 1]
```diff
@@ -156,7 +156,7 @@ const usePlayer = (
       if (playerEl.hasAttribute("autoplay") && !playerEl.hasPlayed) {
         return false;
       }
-      return true;
+      return defaultHasChanged(playerEl, value, propName);
```
Comment (gkatsev) [inline packages/mux-player-react/src/index.tsx:159]: 👍🏻

File: packages/mux-video-react/src/index.tsx [hunk 2]
```diff
@@ -1,11 +1,10 @@
 import useCombinedRefs from "./use-combined-refs";
-import React, { useEffect, useRef, useState } from "react";
+import React, { useEffect, useMemo, useRef, useState } from "react";
```
Comment (gkatsev) [inline packages/mux-video-react/src/index.tsx:2]: looks like `useMemo` isn't used anymore.

File: examples/nextjs-with-typescript/pages/MuxVideo.tsx [hunk 3]
```diff
@@ -1,14 +1,18 @@
...
 
 function MuxVideoPage() {
   const mediaElRef = useRef(null);
-  const [autoplay, setAutoplay] = useState(INITIAL_AUTOPLAY);
+  const [autoplay, setAutoplay] = useState<"muted" | boolean>(INITIAL_AUTOPLAY);
   const [muted, setMuted] = useState(INITIAL_MUTED);
+  const [paused, setPaused] = useState<boolean | undefined>(true);
+  useEffect(() => {
+    if (!mediaElRef.current) return;
+  }, [paused]);
```
Comment (gkatsev) [inline examples/nextjs-with-typescript/pages/MuxVideo.tsx:15]: Does this have some kind of effect or should it be removed?

File: packages/mux-audio-react/src/index.tsx [hunk 4]
```diff
@@ -75,16 +76,13 @@ const MuxAudio = React.forwardRef<HTMLAudioElement | undefined, Partial<Props>>(
...
       );
       playbackEngineRef.current = nextPlaybackEngineRef;
-      const updateAutoplay = setupAutoplay(
-        mediaElRef.current,
-        autoPlay,
-        playbackEngineRef.current
+      setUpdateAutoplay(() =>
+        setupAutoplay(mediaElRef.current, autoPlay, playbackEngineRef.current)
       );
-      updateAutoplayRef.current = updateAutoplay;
```
Comment (gkatsev) [inline packages/mux-audio-react/src/index.tsx:83]: ok, the new way makes more sense and gets around that readonly type issue. I was following the `plabyackEngineRef` code on line 77.

**Review by gkatsev — COMMENTED** [review]
Thanks! I guess I wasn't _too_ far off, but I don't think I could've gotten the last mile by myself.

File: examples/nextjs-with-typescript/pages/MuxVideo.tsx (hunk 3 above)
Comment (cjpillsbury) [reply examples/nextjs-with-typescript/pages/MuxVideo.tsx:15]: lol yeah should be removed. I was stubbing paused toggle but then I realized that `MuxVideo`/`mux-video` doesn't have `paused`.

**Review by gkatsev — APPROVED** [review]

**Pull request merged**